src/cryptonote_core/miner.cpp

```diff
@@ -408,5 +528,313 @@ namespace cryptonote
     return true;
   }
   //-----------------------------------------------------------------------------------------------------
-}
+  bool miner::get_is_background_mining_enabled() const
+  {
+    return m_is_background_mining_enabled;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  bool miner::set_is_background_mining_enabled(const bool& is_background_mining_enabled)
```

Comment:
I recommending removing cases (in this patch) of built-in types being passed by const reference. Pass-by-value instead. Hopefully the remainder of the codebase can be move towards that goal as well ...